sdk/src/binary/system.rs

```diff
@@ -49,4 +52,19 @@ impl<B: BinaryClient> SystemClient for B {
     async fn heartbeat_interval(&self) -> IggyDuration {
         self.get_heartbeat_interval()
     }
+
+    async fn snapshot(
+        &self,
+        compression: SnapshotCompression,
+        snapshot_types: Vec<SystemSnapshotType>,
+    ) -> Result<Snapshot, IggyError> {
+        let response = self
```

Comment:
Let's include here the same line as in the other methods requiring the auth (to fail fast on the client-side).

```
fail_if_not_authenticated(self).await?;
```

Reply:
done